### PR TITLE
Fix bottommost slots position in answer page (#221)

### LINKED ISSUE #221: Bottommost slots on answer page don't render at correct position
I have noticed that slots have a slight shift in positioning the closer they are to the bottom of the timeline. Here's an example:
Screen Shot 2020-09-05 at 19 44 57
23:30 appears just to the right of 24. To my understanding, this slot should have started a little higher, right?

I quickly debugged it and it seems to be related to the fact that the timeline overflows a little bit from the Grid Row, although please dig into more detail when finding a solution

### PR DESCRIPTION
**Pull request by asieraguado**
image

This change makes the hours and answers columns with the same height, fixing the bottommost answers position. I used a header with a white space for having the same fixed-height elements in the hours and answers columns, but I'm not sure if it's the prettiest solution.

closes #221

### CODE REVIEW

File: newdle/client/src/components/answer/Calendar.js [hunk 1]
```diff
@@ -156,19 +156,24 @@ function Hours({minHour, maxHour, hourStep}) {
...
-          style={{top: `${(i / hourSpan) * 100}%`}}
-        >
-          {n === 0 && hourSeries[n] === 0
-            ? '0'
-            : serializeDate(toMoment({hours: hourSeries[n]}), 'k')}
-        </div>
-      ))}
-    </div>
+    <>
+      <Header as="h3" className={styles.date}>
```
Comment (pferreir) [inline newdle/client/src/components/answer/Calendar.js:160]: Is this really necessary? Couldn't we add some padding/margin in `hours-column` instead?

**Review by pferreir — COMMENTED** [review]
Thanks for the PR! I am not sure abour the "empty header". Maybe there's some other way to do that?

**Comment by asieraguado** [comment]
Thank you for your review @pferreir . Replaced the empty header with a different margin property.

**Review by plourenco — COMMENTED** [review]
Thanks @asieraguado ! I was working on another PR and this problem was affecting me too.

However, I noticed that the current height and margin doesn't seem to work very well with hour-long blocks closer to the bottom (24 mid-night).

image

Same result in mobile:

image

Perhaps we can find another balance that works both for half and an-hour blocks?

**Comment by asieraguado** [comment]
> Thanks @asieraguado ! I was working on #230 and this affected me too.
> 
> However, I noticed that the current height and margin doesn't seem to work very well with hour-long blocks closer to the bottom (24 mid-night).
> 
> image
> 
> Same result in mobile:
> 
> image
> 
> Perhaps we can find another balance that works both for half and an-hour blocks?

The height looks correct to me, they would look the same at the top. Maybe you mean that the selection blocks should start and end right in the middle of the hour number? (instead of starting the bottom).

**Review by plourenco — APPROVED** [review]
What I mentioned earlier now seems a bit opinion-based. I've checked with the team and they agree with the current positioning. So, all good on my side!

**Comment by ThiefMaster** [comment]
Now that I've seen it for a time slot starting right at an even hour I'm less sure what's the proper way to align this...

master:
image

your pr:
image

**Comment by ThiefMaster** [comment]
What do you think about adding some thin horizontal lines for each hour? A bit like the ones in indico's room booking calendar, just horizontal instead of vertical:

image

With those lines added we could nicely align the timeslot boxes and it would make any misalignment (like the one this PR fixes) very obvious to notice

**Comment by asieraguado** [comment]
> What do you think about adding some thin horizontal lines for each hour? A bit like the ones in indico's room booking calendar, just horizontal instead of vertical:
> 
> image
> 
> With those lines added we could nicely align the timeslot boxes and it would make any misalignment (like the one this PR fixes) very obvious to notice

Sounds good. I can update the PR with something like this, if you are interested.

**Comment by ThiefMaster** [comment]
i think it's a good idea :)

**Comment by asieraguado** [comment]
Added a new commit to the PR with the hour separators. A timeline with a few slots (some of them overlapping with the separator) would look like this now:

image

**Comment by ThiefMaster** [comment]
I would make the separators wider tan the slots - that way it'll look less strange in case of a slot starting/ending right on top of a separator

**Comment by asieraguado** [comment]
> I would make the separators wider than the slots - that way it'll look less strange in case of a slot starting/ending right on top of a separator

Thanks for the idea, I made the slots a bit narrower so that the separator is wider. This looks better than widening the separator when there are multiple days as below.

image

A few _squeezed_ slots would have a bit less space, but I think it's still usable:

image

**Comment by ThiefMaster** [comment]
Instead of making the slots narrower (I think not even seeing the whole start time isn't amazing), what do you think about making the hour separator span over all three columns instead of having separate ones in each column?

**Comment by asieraguado** [comment]
The previous columns would look like below after the last commit.

image

File: newdle/client/src/components/answer/answer.module.scss [hunk 2]
```diff
@@ -160,6 +163,16 @@
     opacity: 0.3;
     width: 100%;
   }
+
+  .hours-separator {
+    position: absolute;
+    border: 0px;
```
Comment (ThiefMaster) [inline newdle/client/src/components/answer/answer.module.scss:169]: `0` doesn't need a unit (same below for the margin)

**Comment by ThiefMaster** [comment]
Looks really nice now! I'll merge it once you rebased and fixed the conflicts .

**Review by ThiefMaster — APPROVED** [review]

**Pull request merged**